File: app/src/test/java/com/commit451/gitlab/CountingRouter.java
package com.commit451.gitlab;

import android.net.Uri;

import com.commit451.gitlab.navigation.RoutingNavigator;

/**
 * Checks counting
 */
public class CountingRouter implements RoutingNavigator {

    public int issueRouteCount = 0;
    public int commitRouteCount = 0;
    public int mergeRequestRouteCount = 0;
    public int projectRouteCount = 0;
    public int buildRouteCount = 0;
    public int milestoneRouteCount;
    public int unknownRountCount = 0;

    @Override
    public void onRouteToIssue(String projectNamespace, String projectName, String issueIid) {
        issueRouteCount++;
    }

    @Override
    public void onRouteToCommit(String projectNamespace, String projectName, String commitSha) {
        commitRouteCount++;
    }

    @Override
    public void onRouteToMergeRequest(String projectNamespace, String projectName, String mergeRequestId) {
        mergeRequestRouteCount++;
    }

    @Override
    public void onRouteToProject(String namespace, String projectId) {
        projectRouteCount++;
    }

    @Override
    public void onRouteToBuild(String projectNamespace, String projectName, String buildNumber) {
        buildRouteCount++;
    }

    @Override
    public void onRouteToMilestone(String projectNamespace, String projectName, String milestoneNumber) {
        milestoneRouteCount++;
    }

    @Override
    public void onRouteUnknown(Uri uri) {
        unknownRountCount++;
    }
}

File: app/src/androidTest/java/com/commit451/gitlab/CountingRouter.kt
package com.commit451.gitlab

import android.net.Uri
import com.commit451.gitlab.navigation.RoutingNavigator

/**
 * Checks counting
 */
class CountingRouter : RoutingNavigator {

    var issueRouteCount = 0
    var commitRouteCount = 0
    var mergeRequestRouteCount = 0
    var projectRouteCount = 0
    var buildRouteCount = 0
    var milestoneRouteCount: Int = 0
    var unknownRountCount = 0

    override fun onRouteToIssue(projectNamespace: String, projectName: String, issueIid: String) {
        issueRouteCount++
    }

    override fun onRouteToCommit(projectNamespace: String, projectName: String, commitSha: String) {
        commitRouteCount++
    }

    override fun onRouteToMergeRequest(projectNamespace: String, projectName: String, mergeRequestId: String) {
        mergeRequestRouteCount++
    }

    override fun onRouteToProject(namespace: String, projectId: String) {
        projectRouteCount++
    }

    override fun onRouteToBuild(projectNamespace: String, projectName: String, buildNumber: String) {
        buildRouteCount++
    }

    override fun onRouteToMilestone(projectNamespace: String, projectName: String, milestoneNumber: String) {
        milestoneRouteCount++
    }

    override fun onRouteUnknown(uri: Uri?) {
        unknownRountCount++
    }
}
